addons/frontend/index.html

```diff
@@ -15,6 +15,15 @@
     <style id="preview-theme-styles"></style>
     <!-- before all of code to avoid rerender by style replacement -->
     <script>
+      if (typeof acquireVsCodeApi !== "undefined") {
+        /// The string here is a placeholder 
+        /// It will be replaced by the actual preview mode. 
+        let state = `preview-arg:state:{}`;
+        state = state.replace("preview-arg:state:", "");
+        /// Set it later when acquiring the VSCode API
+        window.vscode_state = JSON.parse(state);
```

Comment:
Is it possible if state contains `preview-arg:state:` as a substring? I personnally feel code here is a little bit fragile but i think it's ok at this moment

Reply:
Replace only modifies the first occurence I think:
```js
>>> "xx".replace("x", "y")
"yx"
```

So this shouldn't be an issue (it is indeed fragile tho)